### PR TITLE
using string_view in node::incoming_request and plugin::on_dht_request

### PR DESCRIPTION
**Pull request by aldenml**
I'm a little worried about the performance impact of this change, what I want to do at the end is change the signature of `plugin::on_dht_request`

### CODE REVIEW

**Comment by arvidn** [comment]
I don't see why this would have any worse performance. It's a good change. Is there a reason to hold off on changing the plugin API? btw, that should be mentioned in the Changelog

**Comment by aldenml** [comment]
good, I will go ahead with the plugin API (I wanted a quick review first)

**Comment by aldenml** [comment]
btw, I was worried about the extra conversions of the static string constants during the `==` operator, but I assume the compiler could optimize this cleverly.

**Comment by arvidn** [comment]
I would expect string_view to have a comparison operator against `template<size_t N> char[N]`, making it even more efficient than `strcmp()`. At the very least, I would expect a future version of string_view to have that

File: src/entry.cpp [hunk 1]
```diff
@@ -43,15 +43,11 @@ POSSIBILITY OF SUCH DAMAGE.
...
 
-namespace
-{
-}
-
 namespace libtorrent
 {
 	namespace detail
 	{
-		TORRENT_EXPORT char const* integer_to_str(char* buf, int size
```
Comment (arvidn) [inline src/entry.cpp]: I believe this is used from a public inlined function, isn't it? (in which case it does need to be exported)

Comment (aldenml) [inline src/entry.cpp]: true, about to remove it (btw, it bothers me that this function is only used in `bencode` but declared an defined in `entry`)

Comment (arvidn) [inline src/entry.cpp]: yeah, bencode and entry have an interesting (tight) relationship :p

**Pull request merged**